Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInput.cs

```diff
@@ -658,15 +666,22 @@ private void InitializeActions()
             if (m_Actions == null)
                 return;
 
+            var uiModule = GetComponent<InputSystemUIInputModule>();
+            if (uiModule != null)
+            {
+                if (uiModule.actionsAsset != m_Actions)
+                    uiModule = null;
+            }
+
             ////REVIEW: should we *always* Instantiate()?
             // Check if we need to duplicate our actions by looking at all other players. If any
             // has the same actions, duplicate.
             for (var i = 0; i < s_AllActivePlayersCount; ++i)
                 if (s_AllActivePlayers[i].m_Actions == m_Actions && s_AllActivePlayers[i] != this)
-                {
                     m_Actions = Instantiate(m_Actions);
-                    break;
-                }
```

Comment:
Why remove the break? The first `Instantiate(m_Actions)` would yield a new instance distinct from all other instances so going on with the iteration seems pointless.

Reply:
Yeah, this was unintentional (I refactored this code a little back and forth). Will fix!